### PR TITLE
DOCS align docstrings for linearalgebra module with template

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: dpnp/dpnp_iface_linearalgebra.py [hunk 1]
```diff
@@ -115,6 +99,7 @@ def dot(x1, x2, **kwargs):
 
         if not (dim1 >= 2 and dim2 == 1) and not (dim1 >= 2 and dim2 >= 2) and (x1.dtype == x2.dtype):
             result = dpnp_dot(x1, x2)
+            print("hello")
```
Comment (samir-nasibli) [inline dpnp/dpnp_iface_linearalgebra.py:102]: Please, remove such kind of things before commit. This behavior is not provided by `numpy` API

File: dpnp/dpnp_iface_linearalgebra.py [hunk 2]
```diff
@@ -231,10 +290,11 @@ def vdot(*args, **kwargs):
     """
     Return the dot product of two vectors.
 
+    For full documentation refer to :obj:`numpy.vdot`.
+
     See Also
-    --------
-    :obj:`numpy.vdot`
+    ---------
```
Comment (samir-nasibli) [inline dpnp/dpnp_iface_linearalgebra.py:296]: remove extra sign

File: dpnp/dpnp_iface_linearalgebra.py [hunk 3]
```diff
@@ -192,12 +212,29 @@ def kron(x1, x2, **kwargs):
...
+    >>> import dpnp as np
+    >>> a = np.array([1, 1, 1])
+    >>> b = np.array([1, 2, 3])
+    >>> result = np.outer(a, b)
+    >>> [x for x in result]
+    [1, 2, 3, 1, 2, 3, 1, 2, 3]
 
-    .. seealso:: :obj:`numpy.outer`
+    See Also
+    -------
```
Comment (samir-nasibli) [inline dpnp/dpnp_iface_linearalgebra.py:235]: add missed sign `-`

File: dpnp/dpnp_iface_linearalgebra.py (hunk 1 above)
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:102]: Please remove this line.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 4]
```diff
@@ -63,46 +63,30 @@
...
-        Output argument. This must have the exact kind that would be returned
-        if it was not used. In particular, it must have the right type, must be
-        C-contiguous, and its dtype must be the dtype that would be returned
-        for `dot(a,b)`. This is a performance feature. Therefore, if these
-        conditions are not met, an exception is raised, instead of attempting
-        to be flexible.
+    Limitations
+    -------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+        Arguments ``**kwargs`` is supported only with value `None`.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:73]:
```suggestion
        Keyword arguments ``kwargs`` are currently unsupported.
```
Please apply the change where described `kwargs`.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 5]
```diff
@@ -63,46 +63,30 @@
...
-    out : ndarray, optional
-        Output argument. This must have the exact kind that would be returned
-        if it was not used. In particular, it must have the right type, must be
-        C-contiguous, and its dtype must be the dtype that would be returned
-        for `dot(a,b)`. This is a performance feature. Therefore, if these
-        conditions are not met, an exception is raised, instead of attempting
-        to be flexible.
+    Limitations
+    -------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:72]:
```suggestion
        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 6]
```diff
@@ -63,46 +63,30 @@
...
-        If `out` is given, then it is returned.
+    >>> import dpnp as np
+    >>> np.dot(3, 4)
+    12
+    >>> a = np.array([1, 2, 3])
+    >>> b = np.array([1, 2, 3])
+    >>> np.dot(a, b)
+    14
+
+    See Also
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:86]: Let's have `seealso` before examples everywhere.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 7]
```diff
@@ -63,46 +63,30 @@
...
-        to be flexible.
+    Limitations
+    -------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+        Arguments ``**kwargs`` is supported only with value `None`.
+        Otherwise the functions will be executed sequentially on CPU.
 
-    Returns
+    Examples
     -------
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:77]:
```suggestion
    --------
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 8]
```diff
@@ -161,12 +157,30 @@ def einsum_path(*args, **kwargs):
...
+    Returns the inner product of two arrays.
 
-    The input arrays are flattened into 1-D vectors and then it performs inner
-    product of these vectors.
+    For full documentation refer to :obj:`numpy.inner`
 
-    .. seealso:: :obj:`numpy.inner`
+    Limitations
+    --------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:166]:
```suggestion
        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray`.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 9]
```diff
@@ -183,7 +197,13 @@ def kron(x1, x2, **kwargs):
     """
     Returns the kronecker product of two arrays.
 
-    .. seealso:: :obj:`numpy.kron`
+    For full documentation refer to :obj:`numpy.einsum_path`
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:200]:
```suggestion
    For full documentation refer to :obj:`numpy.kron`
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 10]
```diff
@@ -192,12 +212,29 @@ def kron(x1, x2, **kwargs):
...
-    Returns the outer product of two vectors.
+    Returns the outer product of two arrays.
+    
+    For full documentation refer to :obj:`numpy.outer`
 
-    The input arrays are flattened into 1-D vectors and then it performs outer
-    product of these vectors.
+    Limitations
+    --------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:221]:
```suggestion
        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray`.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 11]
```diff
@@ -214,7 +251,29 @@ def tensordot(x1, x2, axes=2):
...
     Compute tensor dot product along specified axes.
 
-    .. seealso:: :obj:`numpy.tensordot`
+    For full documentation refer to :obj:`numpy.tensordot`
+
+    Limitations
+    --------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+        Arguments ``**kwargs`` is supported only with value `None`.
+        Parameter ``axes`` is supported only with value `1`.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:260]:
```suggestion
        Parameter ``axes`` is supported only with value ``1``.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 12]
```diff
@@ -214,7 +251,29 @@ def tensordot(x1, x2, axes=2):
     """
     Compute tensor dot product along specified axes.
 
-    .. seealso:: :obj:`numpy.tensordot`
+    For full documentation refer to :obj:`numpy.tensordot`
+
+    Limitations
+    --------
+        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:258]:
```suggestion
        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray`.
```

File: dpnp/dpnp_iface_linearalgebra.py (hunk 2 above)
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:296]:
```suggestion
    --------
```

Maybe we will mention that the function works the same as :obj:`dpnp.dot` in `Notes` block. What do you think about that?

File: dpnp/dpnp_iface_linearalgebra.py [hunk 13]
```diff
@@ -159,29 +158,29 @@ def inner(x1, x2, **kwargs):
...
     Returns the inner product of two arrays.
 
-    For full documentation refer to :obj:`numpy.inner`
+    For full documentation refer to :obj:`numpy.inner`.
 
     Limitations
-    --------
-        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:165]: I'm not sure `x1` and `x2` should be of the same type.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 14]
```diff
@@ -214,28 +213,28 @@ def outer(x1, x2, **kwargs):
...
     Returns the outer product of two arrays.
     
-    For full documentation refer to :obj:`numpy.outer`
+    For full documentation refer to :obj:`numpy.outer`.
 
     Limitations
-    --------
-        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:220]: I'm not sure `x1` and `x2` should be of the same type.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 15]
```diff
@@ -251,17 +250,22 @@ def tensordot(x1, x2, axes=2):
...
     Compute tensor dot product along specified axes.
 
-    For full documentation refer to :obj:`numpy.tensordot`
+    For full documentation refer to :obj:`numpy.tensordot`.
 
     Limitations
-    --------
-        Parameters ``x1`` and ``x2`` is supported as :obj:`dpnp.ndarray` and the same type.
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:257]: I'm not sure `x1` and `x2` should be of the same type.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 16]
```diff
@@ -292,8 +291,12 @@ def vdot(*args, **kwargs):
 
     For full documentation refer to :obj:`numpy.vdot`.
 
+    Notes
+    -----
+    This function works the same as :obj:`dpnp.dot`
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:296]:
```suggestion
    This function works the same as :obj:`dpnp.dot`.
```
`Notes` should be located after `See Also` according to the template.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 17]
```diff
@@ -127,16 +111,20 @@ def dot(x1, x2, **kwargs):
...
+
+    Limitations
+    -----------
+    Function is executed sequentially on CPU.
 
     See Also
-    --------
-    :obj:`numpy.einsum`
+    -------
+    :obj:`dpnp.einsum_path`
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:124]: Maybe to add short description of the function?

File: dpnp/dpnp_iface_linearalgebra.py [hunk 18]
```diff
@@ -161,12 +156,30 @@ def einsum_path(*args, **kwargs):
...
+    Returns the inner product of two arrays.
+
+    For full documentation refer to :obj:`numpy.inner`.
 
-    The input arrays are flattened into 1-D vectors and then it performs inner
-    product of these vectors.
+    Limitations
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
+        Arguments ``**kwargs`` is supported only with value `None`.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:166]:
```suggestion
        Keyword arguments ``kwargs`` are currently unsupported.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 19]
```diff
@@ -192,12 +211,29 @@ def kron(x1, x2, **kwargs):
...
+    Returns the outer product of two arrays.
+    
+    For full documentation refer to :obj:`numpy.outer`.
 
-    The input arrays are flattened into 1-D vectors and then it performs outer
-    product of these vectors.
+    Limitations
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray` of the same type.
+        Arguments ``**kwargs`` is supported only with value `None`.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:221]:
```suggestion
        Keyword arguments ``kwargs`` are currently unsupported.
```

File: dpnp/dpnp_iface_linearalgebra.py [hunk 20]
```diff
@@ -214,7 +250,29 @@ def tensordot(x1, x2, axes=2):
...
+    --------
+    :obj:`dpnp.dot` : Returns the dot product.
+    :obj:`dpnp.einsum` : Evaluates the Einstein summation convention on the operands.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    >>> b = np.array([1, 2, 3])
+    >>> np.tensordot(a, b, 1)
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:272]: This line looks excess.

File: dpnp/dpnp_iface_linearalgebra.py [hunk 21]
```diff
@@ -214,7 +250,28 @@ def tensordot(x1, x2, axes=2):
     """
     Compute tensor dot product along specified axes.
 
-    .. seealso:: :obj:`numpy.tensordot`
+    For full documentation refer to :obj:`numpy.tensordot`.
+
+    Limitations
+    -----------
+        Parameters ``x1`` and ``x2`` are supported as :obj:`dpnp.ndarray`.
+        Arguments ``**kwargs`` is supported only with value `None`.
```
Comment (densmirn) [inline dpnp/dpnp_iface_linearalgebra.py:258]:
```suggestion
        Keyword arguments ``kwargs`` are currently unsupported.
```

**Review by densmirn — COMMENTED** [review]
Sorry, I forgot one think:
Let's add the following limitation:
```
Input array data types are limited by supported DPNP :ref:`Data types`.
```
at the end of `Limitations` for the following functions:
`dot`, `inner`, `outer`, `tensordot`

**Review by densmirn — APPROVED** [review]

**Pull request merged**